docs/Killswitches.md

```diff
@@ -1,62 +1,177 @@
+
+Each entry in `kills` must contain at least a `reason` field describing why the
+killswitch was added. EDMC will show this to the user
+(for internal killswitches, anyway).
+
+| Key (* = required) |       Type       | Description                                                                                   |
+| -----------------: | :--------------: | :-------------------------------------------------------------------------------------------- |
+|          `reason`* |      `str`       | The reason that this killswitch was added                                                     |
+|       `set_fields` | `Dict[str, Any]` | A map of key -> contents to update (or overwrite) existing data with                          |
+|    `redact_fields` |   `List[str]`    | A list of traversal paths to redact. This is the same as using set with a value of "REDACTED" |
+|    `delete_fields` |   `List[str]`    | A list of traversal paths in the matching event to be removed, if they exist.                 |
+
+The order listed above is the precedence for actions. i.e. All set fields are
+set, then all redact fields are redacted then all delete fields are deleted.
 
-|       Key |       Type       | Description                                                                  |
-| --------: | :--------------: | :--------------------------------------------------------------------------- |
-| `version` |  `version spec`  | The version of EDMC these kill switches apply to (Must be valid semver spec) |
-|   `kills` | `Dict[str, str]` | The different keys to disable, and the reason for the disable                |
 An example follows:
 
 ```json
 {
-    "version": 1,
-    "last_updated": "19 October 2020",
+    "version": 2,
+    "last_updated": "3 July 2021",
     "kill_switches": [
         {
             "version": "1.0.0",
             "kills": {
-                "plugins.eddn.send": "some reason"
+                "plugins.eddn.send": {
+                    "reason": "some reason",
+                    "delete_fields": ["world_domination_plans"],
+                    "set_fields": {
+                        "bad_bases_for_systems_of_government": ["Strange women lying in ponds distributing swords"],
+                        "ruler_map": {"emperor": "scimitar"}
+                    },
+                    "redact_fields": ["relation_to_thargoids"]
+                },
+                "plugins.some_plugin.some_thing": {
+                    "reason": "Some thing is disabled pending investigation of NMLA relations."
+                }
             }
         }
     ]
 }
 ```
 
+- `plugins.edsm.send` will have fields deleted, set, and redacted, and then
+  will *not* be halted, the send will continue with the modified data.
+- `plugins.some_plugin.some_thing` will never be allowed to continue
+  (as all fields are blank)
+
+Indexing of lists (and any other `Sequence`) is done with numbers as normal.
+Negative numbers do work to reference the end of sequences.
```

Comment:
As I went searching yesterday, add a note that:

```
To append you will need to use `len(array/set/whatever)`, there's no way to do it just with indexing.
```

I wish `-0` was special in that way, but obviously all of python's parsing would need the concept of 'negative zero' for that to work.

Reply:
Ah never mind, you covered this below in the Sets section.